account_fiscal_position_vies_warning/models/account_move.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2025 Moduon Team S.L. <info@moduon.team>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.exceptions import ValidationError
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def action_post(self):
+        if self.env.company.vat_check_vies:  # Verify VAT Numbers is True
+            company_id = self.env.company.id
+            try:
+                if (
+                    self.fiscal_position_id
```

Comment:
You're assuming a single record logic, but the main method works over recordsets. And thinking about this, if different companies invoices are posted in the method (maybe some purchase-sale intercompany logic), you should be looking the `vat_check_vies` check for everyone, not just the active one...